deps/obl/oblas_lite.c

```diff
@@ -1,6 +1,43 @@
 #include "oblas_lite.h"
 #include "gf2_8_mul_table.h"
 
+#if defined(OBLAS_TINY)
+static uint8_t gf2_8_mul(uint16_t a, uint16_t b)
+{
+    uint16_t acc = 0;
+
+    // Perform 8-bit, carry-less multiplication of |a| and |b|.
+    if (a && b) {
+        acc = GF2_8_EXP[GF2_8_LOG[a] + GF2_8_LOG[b]];
+    }
+
+    // Add multiples of the modulus to eliminate all bits past a byte. Note that
+    // the bits in |modulus| have a one where there's a non-zero power of |x| in
+    // the field modulus.
+    uint16_t modulus = 0b100011101 << 7;
+    for (int i = 15; i >= 8; i--) {
+        const uint16_t mask = ~((acc >> i) - 1);
+        acc ^= modulus & mask;
+        modulus >>= 1;
+    }
+
+    return acc;
+}
+
+static void obl_axpy_ref(u8 *a, u8 *b, u8 u, unsigned k)
+{
+    register u8 *ap = a, *ae = &a[k], *bp = b;
+    for (; ap != ae; ap++, bp++)
+        *ap ^= gf2_8_mul(u, *bp);
+}
+
+static void obl_scal_ref(u8 *a, u8 *b, u8 u, unsigned k)
+{
+    register u8 *ap = a, *ae = &a[k];
+    for (; ap != ae; ap++)
+        *ap = gf2_8_mul(u, *ap);
+}
+#else
```

Comment:
Would you consider this simpler change? The modulus/accumulator bits are not necessary when using the log and exp tables, and `gf2_8_mul()` can simply be a macro. Finally the multiplication lookup table is only included if `OBLAS_TINY` is not defined.

```suggestion
#if defined(OBLAS_TINY)
#define GF2_8_MUL(a, b) (((a) && (b)) ? GF2_8_EXP[GF2_8_LOG[(a)] + GF2_8_LOG[(b)]] : 0)

static void obl_axpy_ref(u8 *a, u8 *b, u8 u, unsigned k)
{
    register u8 *ap = a, *ae = &a[k], *bp = b;
    for (; ap != ae; ap++, bp++)
        *ap ^= GF2_8_MUL(u, *bp);
}

static void obl_scal_ref(u8 *a, u8 *b, u8 u, unsigned k)
{
    register u8 *ap = a, *ae = &a[k];
    for (; ap != ae; ap++)
        *ap = GF2_8_MUL(u, *ap);
}
#else
#include "gf2_8_mul_table.h"

```